### PR TITLE
minor: make private class final with default constructor

### PR DESCRIPTION
**Pull request by Kevin222004**
minor: make private class final with default constructor

This is part of https://github.com/checkstyle/checkstyle/pull/12737

According to the https://docs.oracle.com/javase/specs/jls/se17/html/jls-8.html#jls-8.8.9 The default constructor has the same access modifier as the class.
and according to the check https://checkstyle.org/config_design.html#FinalClass a class that has only private constructors and has no descendant classes is declared as final.

### CODE REVIEW

**Comment by Kevin222004** [comment]
I have given information on this changes in description. 

can we have your valuable feedback on this changes.

**Comment by hazendaz** [comment]
@Kevin222004 I'm taking a look, in future though, there is no reason to open a PR then immediately ask for an update on it.  We get emailed on these and it just sends needless emails doing that.  Its better to just open it, describe it well and leave it at that.  Thanks.

**Review by hazendaz — APPROVED** [review]

**Comment by Kevin222004** [comment]
@hazendaz we need your feedback on these changes.

**Comment by Kevin222004** [comment]
What do you think should we care about this thing or not

**Comment by hazendaz** [comment]
I merged this an hour ago.  Please make sure to follow github here and what is happening.  My first comment was to assist in not unnecessarily sending spam comments.  You have followed that by two more needless spam emails.

To answer your question yes this is valid.  Code should be written properly.  Many fail to do so on most if not all repos thinking java is too wordy.  I personally don't agree with ignoring such code requirements in java and it typically seems silly that even a library such as this doesn't fully follow all coding conventions.  Tools like open rewrite would have likely applied the same here, same with Eclipse / Intellij, etc.  You changes were otherwise accurate and have been merged.
